### PR TITLE
[1.10.0 Release] Add release logs for 1.10.0

### PR DESCRIPTION
**Pull request by architkulkarni**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
@jjyao could you please check this for regressions?  https://github.com/ray-project/ray/pull/21908/commits/da0593fde94676b80a21487bcd555bf309e4cb23

File: release/release_logs/1.10.0/microbenchmark.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+  "1_1_actor_calls_async": [
+    6488.679319380362,
+    239.08621203545292
+  ],
+  "1_1_actor_calls_concurrent": [
+    5369.625993528198,
+    120.17942920626632
+  ],
+  "1_n_actor_calls_async": [
+    11741.041010762125,
```
Comment (jjyao) [inline release/release_logs/1.10.0/microbenchmark.json:67]: This seems to be slower: it was `15614.656699877709` in 1.9

File: release/release_logs/1.10.0/microbenchmark.json [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    485.03043177404095
+  ],
+  "n_n_actor_calls_async": [
+    37042.17466054547,
```
Comment (jjyao) [inline release/release_logs/1.10.0/microbenchmark.json:71]: This might be slower as well.

File: release/release_logs/1.10.0/microbenchmark.json [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+  "1_1_async_actor_calls_async": [
+    3934.6805285223663,
+    388.6803752134044
+  ],
+  "1_1_async_actor_calls_with_args_async": [
+    2804.1953806090437,
+    37.51052431976944
+  ],
+  "1_n_async_actor_calls_async": [
+    12122.735371427163,
```
Comment (jjyao) [inline release/release_logs/1.10.0/microbenchmark.json:91]: This is slower as well.

File: release/release_logs/1.10.0/microbenchmark.json [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+    221.28898311698475
+  ],
+  "n_n_async_actor_calls_async": [
+    33142.40565675831,
```
Comment (jjyao) [inline release/release_logs/1.10.0/microbenchmark.json:95]: Seems actor call is slower in general.

File: release/release_logs/1.10.0/stress_tests/stress_test_dead_actors.json [hunk 5]
```diff
@@ -0,0 +1,12 @@
+{
+  "success": 1,
+  "total_time": 162.9751513004303,
+  "avg_iteration_time": 1.6297491025924682,
+  "max_iteration_time": 6.213320970535278,
+  "min_iteration_time": 0.023737668991088867,
+  "smoke_test": true,
```
Comment (jjyao) [inline release/release_logs/1.10.0/stress_tests/stress_test_dead_actors.json:7]: I think you pasted the wrong one: this is the smoke test.

File: release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+  "stage_1_max_iteration_time": 1.789336919784546,
+  "stage_1_min_iteration_time": 0.8084619045257568,
+  "stage_2_time": 1.0962693691253662,
+  "stage_2_avg_iteration_time": 0.21914191246032716,
+  "stage_2_max_iteration_time": 0.2479391098022461,
+  "stage_2_min_iteration_time": 0.20914483070373535,
+  "stage_3_creation_time": 0.007324695587158203,
+  "stage_3_time": 5.459118604660034,
+  "stage_4_spread": 0.9857997828000066,
+  "smoke_test": true,
```
Comment (jjyao) [inline release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:15]: Same, this is smoke test.

**Comment by architkulkarni** [comment]
Sorry about that, let me update with the non-smoke test.

**Comment by architkulkarni** [comment]
Updated with non-smoke test, sorry for the mistake!

**Comment by architkulkarni** [comment]
@jjyao Thanks for checking these.  Are the regressions release-blocking?  Let me know if it would be helpful for me to get input from someone else.  

For more data, here are the logs for `1.10.0` prior to the last round of cherry picks (runtime env serialization/deserialization and CI fixes): https://github.com/ray-project/ray/pull/21512

Here are the logs for `1.10.0rc0`: https://github.com/ray-project/ray/pull/21434

File: release/release_logs/1.10.0/microbenchmark.json (hunk 1 above)
Comment (architkulkarni) [reply release/release_logs/1.10.0/microbenchmark.json:67]: Adding more details: in `1.10.0` prior to the last round of cherry-picks, this number was `10247.98729565172`.
Int `1.10.0rc0`, it was `10403.857830375695`.

File: release/release_logs/1.10.0/microbenchmark.json (hunk 2 above)
Comment (architkulkarni) [reply release/release_logs/1.10.0/microbenchmark.json:71]: 1.9: `46044`
1.10.0 prior to last round of cherry picks: `31276`
1.10.0rc0: `31721`

**Comment by jjyao** [comment]
Seems the regression is between b743f514d775cec66924cffdd4092a5e767fe310 and 8a943e8081ff4ee505d0fae3bdbb75ff208821de.

**Comment by architkulkarni** [comment]
Thanks for narrowing it down @jjyao!  That range implies it would be a different offending commit than the one for the [stress_test_many_tasks regression](https://github.com/ray-project/ray/issues/21760), which is interesting.

Based on the numbers added above (e.g. https://github.com/ray-project/ray/pull/21908#discussion_r793112085), it looks like the patch here https://github.com/ray-project/ray/pull/21788 (which was added in the last round of cherry-picks) fixed the regression a little less than halfway (very roughly), but it's still a regression.

@zhe-thoughts @ericl @rkooo567 Is this severe enough to block the release?

**Comment by jjyao** [comment]
The slowdown was introduced by 72bd2d7e09badebb998e442f2bae581b56f5a71c (https://buildkite.com/ray-project/periodic-ci/builds/2535#_)

**Comment by ericl** [comment]
Is there a way we can flag off that functionality?

File: release/release_logs/1.10.0/microbenchmark.json (hunk 1 above)
Comment (zhe-thoughts) [reply release/release_logs/1.10.0/microbenchmark.json:67]: Does this number mean the max number of calls that can be finished within unit time? @jjyao @architkulkarni

Comment (jjyao) [reply release/release_logs/1.10.0/microbenchmark.json:67]: This means the number of calls per second (throughput) so the higher the better.

**Comment by mwtian** [comment]
Since #21923 is no longer a release blocker, is there any other issue?

**Review by jjyao — APPROVED** [review]
LGTM

**Comment by architkulkarni** [comment]
Yes, this is the final wheel for 1.10.0 so we should merge this PR.  Merging this PR doesn't block the release.  I don't have merge privileges, perhaps @jjyao or @rkooo567 can merge this?

**Comment by jjyao** [comment]
Neither do I. @ericl or @scv119

**Pull request merged**